metals/src/main/scala/scala/meta/internal/metals/ConfiguredLanguageClient.scala

```diff
@@ -82,19 +82,21 @@ final class ConfiguredLanguageClient(
     }
   }
 
+  override def refreshModel(): CompletableFuture[Unit] = {
+    if (clientConfig.codeLenseRefreshSupport)
+      underlying.refreshCodeLenses.thenApply(_ => ())
+    else if (
+      clientConfig.isExecuteClientCommandProvider && clientConfig.isDebuggingProvider
+    ) {
+      val params = ClientCommands.RefreshModel.toExecuteCommandParams()
+      CompletableFuture.completedFuture(metalsExecuteClientCommand(params))
+    } else CompletableFuture.completedFuture(())
+  }
```

Comment:
Ok, I realized that `refreshModel` in vscode is also used for `test explorer` so current logic breaks its behavior. 
Following should work in both cases:
```scala
    if(clientConfig.isExecuteClientCommandProvider && clientConfig.isDebuggingProvider) {
      val params = ClientCommands.RefreshModel.toExecuteCommandParams()
      metalsExecuteClientCommand(params)
    }

    if (clientConfig.isDebuggingProvider && clientConfig.codeLenseRefreshSupport)
      underlying.refreshCodeLenses.thenApply(_ => ())
    else CompletableFuture.completedFuture(())
  ```
I also added `clientConfig.isDebuggingProvider` to the condition since code lenses are implemented via the DAP server.

Reply:
You mean `metals-model-refresh` shouldn't be deprecated and it still serves a different purpose than `workspace/codeLens/refresh` ?